app/components/ChatHeader.tsx

```diff
@@ -2,19 +2,37 @@ import React from 'react';
 
 interface ChatHeaderProps {
   onClose: () => void;
+  onMaximize: () => void;
+  isMaximized: boolean;
 }
 
-const ChatHeader: React.FC<ChatHeaderProps> = ({ onClose }) => {
+const ChatHeader: React.FC<ChatHeaderProps> = ({ onClose, onMaximize, isMaximized }) => {
   return (
-    <div className="absolute top-2 right-0 z-50 pr-4">
+    <div className="sticky top-0 right-0 z-50 flex justify-end gap-2 bg-[var(--background)] px-4 py-2">
+      <button
+        onClick={onMaximize}
+        className="hidden md:inline-flex p-2 cursor-pointer hover:bg-gray-200 dark:hover:bg-gray-700 rounded w-8 h-8"
+        aria-label={isMaximized ? 'Minimize' : 'Maximize'}
+        aria-pressed={isMaximized}
+      >
+        {isMaximized ? (
+          <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" strokeWidth="2" strokeLinecap="round" strokeLinejoin="round" className="lucide lucide-minimize-icon lucide-minimize">
+            <path d="M8 3v3a2 2 0 0 1-2 2H3" /><path d="M21 8h-3a2 2 0 0 1-2-2V3" /><path d="M3 16h3a2 2 0 0 1 2 2v3" /><path d="M16 21v-3a2 2 0 0 1 2-2h3" />
+          </svg>
+        ) : (
+          <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" strokeWidth="2" strokeLinecap="round" strokeLinejoin="round" className="lucide lucide-maximize-icon lucide-maximize">
+            <path d="M8 3H5a2 2 0 0 0-2 2v3" /><path d="M21 8V5a2 2 0 0 0-2-2h-3" /><path d="M3 16v3a2 2 0 0 0 2 2h3" /><path d="M16 21h3a2 2 0 0 0 2-2v-3" />
+          </svg>
+        )}
+      </button>
+
       <button
         onClick={onClose}
-        className="p-2 cursor-pointer"
+        className="p-2 cursor-pointer hover:bg-gray-200 dark:hover:bg-gray-700 rounded w-8 h-8"
         aria-label="Close"
       >
-        <svg xmlns="http://www.w3.org/2000/svg" className="h-5 w-5 text-gray-600 dark:text-gray-400" viewBox="0 0 20 20" fill="currentColor">
-          <path fillRule="evenodd" d="M4.293 4.293a1 1 0 011.414 0L10 8.586l4.293-4.293a1 1 0 111.414 1.414L11.414 10l4.293 4.293a1 1 0 01-1.414 1.414L10 11.414l-4.293 4.293a1 1 0 01-1.414-1.414L8.586 10 4.293 5.707a1 1 0 010-1.414z" clipRule="evenodd" />
-        </svg>
+        <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" strokeWidth="2" strokeLinecap="round" strokeLinejoin="round" className="lucide lucide-minus-icon lucide-minus">
```

Comment:
Let's also change the close button from using `svg` to use a component from `lucide-react` library